AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Sample_Editor_Tests.py

```diff
@@ -0,0 +1,143 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Test Case: Creating tests for basic editor feature like creating a level, adding new entities, modifying the entities etc.
+"""
+
+
+class Tests:
+    create_level = ("Level created and loaded successfully", "Failed to create the level")
+    open_level = ("Level loaded successfully", "Failed to load the level")
+    create_entity = ("Parent entity created successfully", "Failed to create a parent entity")
+    set_entity_name = ("Entity name set successfully", "Failed to set entity name")
+    delete_entity = ("Parent entity deleted successfully", "Failed to delete parent entity")
+    game_mode_enter = ("Game Mode entered successfully", "Failed to enter the game mode")
+    game_mode_exit = ("Game mode exited successfully", "Failed to exit game mode")
+    create_child_entity = ("Child entity created successfully", "Failed to create a child entity")
+    delete_child_entity = ("Child entity deleted successfully", "Failed to delete child entity")
+    add_mesh_component = ("Mesh component added successfully", "Failed to add mesh component")
+    found_component_typeId = ("Found component typeId", "Unable to find component TypeId")
+    remove_mesh_component = ("Mesh component removed successfully", "Failed to remove mesh component")
+
+
+def sample_editor_tests():
+    """
+        Performing basic test in editor
+            01. create_level if it does not exist else 02. open exiting level
+
+            03. create parent entity and set name
+            04. create child entity
+            05. delete child entity
+            06. add mesh component
+            07. remove mesh component
+            08. enter game mode
+            09. exit game_mode
+            10. delete parent entity
+            11. save level
+
+    """
+    import os
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    import editor_python_test_tools.hydra_editor_utils as hydra
+
+    import azlmbr.math as math
+    import azlmbr.asset as asset
+    import azlmbr.bus as bus
+    import azlmbr.editor as editor
+    import azlmbr.entity as entity
+    import azlmbr.legacy.general as general
+    import azlmbr.object
+
+    def search_entity(entity_to_search, entity_name):
```

Comment:
something similar in function may already exist in .\o3de\AutomatedTesting\Gem\PythonTests\EditorPythonTestTools\editor_python_test_tools\editor_entity_utils.py as def find_editor_entity(cls, entity_name: str) -> EditorEntity:
that returns an EditorEntity object

the params and function seem confusing to me so this takes both an entity name and an entity id as params?
this searches for entities named entity_name then tests if entity_to_search id is on the returned list and returns true/false...
would it be easier to use the entity Id to ask what the name of that entity is and compare to expected? in editor_entity_utils.py you can call get_name() on an entity which internally uses editor.EditorEntityInfoRequestBus(bus.Event, "GetName", self.id).

Reply:
agreed and fixed by using editor utils